include/shark/ObjectiveFunctions/AbstractObjectiveFunction.h

```diff
@@ -83,9 +87,15 @@ class AbstractObjectiveFunction :  public INameable{
 	typedef PointType SearchPointType;
 	typedef ResultT ResultType;
 
-	typedef SearchPointType FirstOrderDerivative;
+	//if the result type is not an arithmetic type, we assume it is a vector-type->multi objective optimization
+	typedef typename boost::mpl::if_<
+		std::is_arithmetic<ResultT>,
+		SearchPointType,
```

Comment:
Hi,

this does not really hurt being there and it allows us in the future to look at different differentiable spaces, e.g. complex.
________________________________
From: Tobias Glasmachers [notifications@github.com]
Sent: Friday, June 03, 2016 11:07 PM
To: Shark-ML/Shark
Cc: Oswin Krause; Author
Subject: Re: [Shark-ML/Shark] Multi objective function (#77)


In include/shark/ObjectiveFunctions/AbstractObjectiveFunction.h<https://github.com/Shark-ML/Shark/pull/77#discussion_r65775592>:

> @@ -83,9 +87,15 @@ class AbstractObjectiveFunction :  public INameable{
>       typedef PointType SearchPointType;
>       typedef ResultT ResultType;
>
> -     typedef SearchPointType FirstOrderDerivative;
> +     //if the result type is not an arithmetic type, we assume it is a vector-type->multi objective optimization
> +     typedef typename boost::mpl::if_<
> +             std::is_arithmetic<ResultT>,
> +             SearchPointType,


If would look more natural to me to replace SearchPointType with RealVector. Well, this is probably not the place for a CompressedRealVector, and a derivative does not make any sense on a discrete search space.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Shark-ML/Shark/pull/77/files/b60b8cd7c437482d7f55c5cb229f72e6fd01712f#r65775592>, or mute the thread<https://github.com/notifications/unsubscribe/AOWTBm1ChsoTtHA_dlbL4k6Ob46UTCpWks5qIJePgaJpZM4IpCNs>.